cdap-client/src/main/java/co/cask/cdap/client/MetaClient.java

```diff
@@ -54,4 +59,23 @@ public Version getVersion() throws IOException, UnauthorizedException {
     HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL("version"), config.getAccessToken());
     return ObjectResponse.fromJsonBody(response, Version.class).getResponseObject();
   }
+
+  public Map<String, ConfigEntry> getCDAPConfig() throws IOException, UnauthorizedException {
+    return getConfig("config/cdap");
+  }
+
+  public Map<String, ConfigEntry> getHadoopConfig() throws IOException, UnauthorizedException {
+    return getConfig("config/hadoop");
+  }
+
+  private Map<String, ConfigEntry> getConfig(String url) throws IOException, UnauthorizedException {
+    HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL(url), config.getAccessToken());
+    List<ConfigEntry> responseObject =
```

Comment:
Yea, I feel that the performance improvement gained by doing that isn't worth doubling the lines of code for the functionality. (Especially when its in clients, where there's so much time spent on network overhead, and performance isn't otherwise critical).
What do you think?